src/base/bittorrent/session.cpp

```diff
@@ -1901,15 +1917,25 @@ bool Session::deleteTorrent(const QString &hash, bool deleteLocalFiles)
     // Remove it from session
     if (deleteLocalFiles) {
         QString rootPath = torrent->rootPath(true);
-        if (!rootPath.isEmpty())
+        QString forcePath;
+        if (!rootPath.isEmpty()) {
             // torrent with root folder
             m_removingTorrents[torrent->hash()] = {torrent->name(), rootPath, deleteLocalFiles};
-        else if (torrent->useTempPath())
+            forcePath = rootPath;
+        }
+        else if (torrent->useTempPath()) {
             // torrent without root folder still has it in its temporary save path
             m_removingTorrents[torrent->hash()] = {torrent->name(), torrent->savePath(true), deleteLocalFiles};
-        else
+            forcePath = torrent->savePath(true);
+        }
+        else {
             m_removingTorrents[torrent->hash()] = {torrent->name(), "", deleteLocalFiles};
+        }
         m_nativeSession->remove_torrent(torrent->nativeHandle(), libt::session::delete_files);
+        if (deleteForce && !forcePath.isEmpty()) {
+            qDebug("Force removing path: %s", qUtf8Printable(forcePath));
+            Utils::Fs::removeDirRecursive(forcePath);
```

Comment:
The sequence is wrong here:
`m_removingTorrents` is a placeholder for torrents to line up for deletion, and you called `removeDirRecursive()` here will go before the processing of `m_removingTorrents` (which is at handleTorrentRemovedAlert & handleTorrentDeletedAlert).
IMO the correct way is calling `removeDirRecursive()` after processing `m_removingTorrents`.
Pardon I couldn't give suggestion how to change it exactly, maybe @glassez share his opinion.

Reply:
Yes. Here we store some info about torrent that will be deleted and ask libtorrent to delete it. All cleanups are performed in corresponding libtorrent alert handlers.

Reply:
I agree all existing cleanups are performed in libtorrent... but libtorrent doesn't seem to have a force function, so i can't really do this forced cleanup using libtorrent functionality... that's why i put it after the "remove_torrent" call which reaches out to libtorrent and does its deletes... maybe i'm wrong, but the basic logic here is:

"remove_torrent" call uses libt to do all the deletes that it can
then 
if "deleteForce" is true and the path is not empty, use qbt's "removeDirRecursive" to clean up everything else... 

I realize this could be accomplished with a single delete call, but it seemed better to let libt do its thing, and then follow that up (force) with qbt specific code to finish deleting anything libt didn't handle... rather then to override or modify libt functions in this case... specifically because it is a "forcing" operation we're implementing here, not a standard operating procedure supported by libtorrent I wanted to preserve the standard functionality of libt in this code section as much as possible...

I've left this code as i originally made it in the latest PR updates, however i'm open to further discussion about what you were thinking might work better in this case (maybe i'm not understanding what you're trying to get me to do with this code)...